File: examples/ios/CMeSdkObjc/CMeSdkObjc/CustomTableViewCell.m
//
//  CustomTableViewCell.m
//  CMeSdkObjc
//
//  Created by Evernym on 01.07.2021.
//  Copyright © 2021 Norman Jarvis. All rights reserved.
//

#import "CustomTableViewCell.h"
typedef void (^ActionBlock)(void);

@interface CustomTableViewCell ()

@property ActionBlock _acceptBlock;
@property ActionBlock _rejectBlock;

@end

@implementation CustomTableViewCell
@synthesize _acceptBlock, _rejectBlock;

- (void)awakeFromNib {
    [super awakeFromNib];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];
}

-(void) callAcceptBlock:(id)sender {
    _acceptBlock();
}

-(void) callRejectBlock:(id)sender {
    _rejectBlock();
}

- (void)updateAttribute: (NSString*) title
               subtitle: (NSString*) subtitle
                logoUrl: (NSString*) logoUrl
                 acceptCallback: (ActionBlock) acceptCallback
         rejectCallback: (ActionBlock) rejectCallback {
    self.title.text = title;
    self.subtitle.text = subtitle;
    
    NSURL *url = [NSURL URLWithString:logoUrl];
    NSData *data = [[NSData alloc] initWithContentsOfURL:url];
    UIImage *image = [UIImage imageWithData:data];
    self.logo.image = image;
    
    _acceptBlock = acceptCallback;
    [self.accept addTarget:self action:@selector(callAcceptBlock:) forControlEvents:UIControlEventTouchUpInside];
    [self.accept showsTouchWhenHighlighted];

    _rejectBlock = rejectCallback;
    [self.reject addTarget:self action:@selector(callRejectBlock:) forControlEvents:UIControlEventTouchUpInside];
    [self.reject showsTouchWhenHighlighted];
}

@end

File: examples/ios/MSDKSampleAppSwift/MSDKSampleAppSwift/CustomTableViewCell.swift
//
//  CustomTableViewCell.swift
//  MSDKSampleAppSwift
//
//  Created by Evernym on 07.07.2021.
//  Copyright © 2021 Norman Jarvis. All rights reserved.
//

class CustomTableViewCell: UITableViewCell {
    @IBOutlet var logo: UIImageView!;
    @IBOutlet var title: UILabel!;
    @IBOutlet var subtitle: UILabel!;
    @IBOutlet var accept: UIButton!;
    @IBOutlet var reject: UIButton!;
    
    typealias ActionBlock = () -> Void;
    var _acceptCallback: ActionBlock? = nil;
    var _rejectCallback: ActionBlock? = nil;
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
    }
    
    @objc func callAcceptBlock() {
        _acceptCallback!();
    }
    
    @objc func callRejectBlock() {
        _rejectCallback!();
    }
    
    func updateCellAttributes(title: String, subtitle: String, logoUrl: String) {
        self.title.text = title;
        self.subtitle.text = subtitle;
        
        if logoUrl != "" {
            let url = URL(string: logoUrl)
            let data = try? Data(contentsOf: url!)
            self.logo.image = UIImage(data: data!)
        }
    }
    
    func addAceptCallback(acceptCallback: @escaping ActionBlock) {
        _acceptCallback = acceptCallback;
        self.accept.addTarget(self, action: #selector(callAcceptBlock), for: UIControl.Event.touchUpInside)
    }
    
    func addRejectCallback(rejectCallback: @escaping ActionBlock) {
        _rejectCallback = rejectCallback;
        self.reject.addTarget(self, action: #selector(callRejectBlock), for: UIControl.Event.touchUpInside)
    }
    
    func loadImage(_ logoPath: String){
        guard
            let logoUrl = URL(string: logoPath)
        else {
            return
        }

        self.logo.isHidden = false;
        logo.layer.cornerRadius = 30
        DispatchQueue.global(qos: .userInitiated).async {
            let data = try? Data(contentsOf: logoUrl)
            guard let imgData = data else { return }

            DispatchQueue.main.async {
                self.logo.image = UIImage(data: imgData)
            }
        }
    }
    
}
